app/services/publisher_wallet_setter.rb

```
@@ -18,18 +18,38 @@ def perform
 
     # This raises when response is not 2xx.
     response = connection.put do |request|
-      request.body =
-          <<~BODY
+      request.headers["Authorization"] = api_authorization_header
+      request.headers["Content-Type"] = "application/json"
+
+      if publisher.publication_type == :site
+        request.body =
+            <<~BODY
             {
               "provider": "uphold", 
               "parameters": #{JSON.dump(uphold_access_parameters)}, 
               "verificationId": "#{publisher.id}"
             }
-          BODY
-      request.headers["Authorization"] = api_authorization_header
-      request.headers["Content-Type"] = "application/json"
-      request.url("/v2/publishers/#{publisher.brave_publisher_id}/wallet")
+        BODY
+        request.url("/v2/publishers/#{publisher.brave_publisher_id}/wallet")
+      elsif publisher.publication_type == :youtube_channel
+        request.body =
+            <<~BODY
+            {
+              "provider": "uphold", 
+              "parameters": #{JSON.dump(uphold_access_parameters)} 
+            }
+        BODY
+        request.url("/v1/owners/#{URI.escape(publisher.owner_identifier)}/wallet")
+      else
+        Rails.logger.warn("PublisherWalletSetter can't set wallet for publication_type #{publisher.publication_type.to_s}")
```

same as the getter; let's also raise an exception and send it to Sentry